components/list/list.js

```diff
@@ -415,7 +415,17 @@ class List extends PageableMixin(SelectionMixin(LitElement)) {
 					e.stopPropagation();
 					target.dispatchSetChildCurrentEvent(true);
 				}, { once: true });
+
 				prevCurrent.current = false;
+				const firstChild = this.querySelector('[_has-current-parent]');
+				if (firstChild) firstChild._hasCurrentParent = false;
+
+				const prevSiblings = this.querySelectorAll('[_next-sibling-current]');
+				if (prevSiblings.length > 0) {
+					prevSiblings.forEach(sibling => {
+						sibling._nextSiblingCurrent = false;
+					});
+				}
```

Comment:
This cleans up `_has-current-parent` and `_next-sibling-current` when a new item is selected.